packages/http-client/src/http-client.ts

```diff
@@ -121,6 +122,32 @@ export function execute<ReturnT>(executeFn: ExecuteHttpRequestFn<ReturnT>) {
   };
 }
 
+function encodeRequestQueryParameters(
+  parameter: Record<string, string> | undefined
+): Record<string, string> | undefined {
+  if (parameter) {
+    return Object.keys(parameter).reduce(
+      (encodedParams, key) => ({
+        ...encodedParams,
+        [key]: encodeURIComponent(parameter[key])
+      }),
+      {}
+    );
+  }
+}
+function getEncodedParameters(
+  parameters: OriginOptionsInternal
+): OriginOptionsInternal {
+  return {
+    custom: parameters.custom,
+    requestConfig: parameters.requestConfig, // sdk parameters are encoded
+    destinationProperty: encodeRequestQueryParameters(
+      parameters.destinationProperty
+    ),
+    destination: encodeRequestQueryParameters(parameters.destination)
```

Comment:
```suggestion
    ...parameters, // sdk parameters are encoded
    destinationProperty: encodeRequestQueryParameters(
      parameters.destinationProperty
    ),
    destination: encodeRequestQueryParameters(parameters.destination)
```
This would be slightly shorter... probably no difference otherwise.

Reply:
good point I adjusted the comment that custom should not be encoded and sdk parameters are encoded.